synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -294,25 +295,32 @@ async def _do_preview(self, url: str, user: UserID, ts: int) -> bytes:
                 # Check if this HTML document points to oEmbed information and
                 # defer to that.
                 oembed_url = self._oembed.autodiscover_from_html(tree)
-                og = {}
+                og_from_oembed: JsonDict = {}
                 if oembed_url:
                     oembed_info = await self._download_url(oembed_url, user)
-                    og, expiration_ms = await self._handle_oembed_response(
+                    (
+                        og_from_oembed,
+                        author_name,
+                        expiration_ms,
+                    ) = await self._handle_oembed_response(
                         url, oembed_info, expiration_ms
                     )
 
-                # If there was no oEmbed URL (or oEmbed parsing failed), attempt
-                # to generate the Open Graph information from the HTML.
-                if not oembed_url or not og:
-                    og = parse_html_to_open_graph(tree, media_info.uri)
+                og_from_og = parse_html_to_open_graph(tree, media_info.uri)
+
+                # If there was no oEmbed URL, or oEmbed parsing failed, or the
+                # information retrieved was incomplete, we complete it from
+                # the OpenGraph information. We give oEmbed information
+                # precedence.
```

Comment:
This is all Open Graph information, I think this wants to say "we complete it by scraping information from the HTML" -- that information may or may not be in `og` tags.

Maybe update this comment to:

```suggestion
                # Compile the Open Graph response by using the scraped
                # information from the HTML and overlaying any information
                # from the oEmbed response.
```

And add a comment the call to `parse_html_to_open_graph` which says:

> Parse Open Graph information from the HTML in case the oEmbed response failed or is incomplete.